Equipment actions

VanAnderson opened the pull request:
This is basically piggy-backing off of the work that Dane did on horse actions. I have some ideas to refactor that I would like to try once we have the bare bones together.

jah2488 commented:
Awesome. Gonna review this in a few

VanAnderson commented:
Sounds good. One thing I was itching to pull into the application controller from the Horses Controller and the Equipment controller was restrict_non_admin. However, it needs the path of the index  for the current controller to redirect. Is there a way to infer this path from the controller? 

I also saw you could pass in arguments to before_action callbacks using a lambda (http://stackoverflow.com/questions/19260288/rails-4-before-action-pass-parameters-to-invoked-method). This could solve the issue by just having the redirect path passed in as a parameter, but it looked a bit messy-- I wasn't sure if it was kosher.

jah2488 commented on app/views/equipment/edit.slim:
```diff
@@ -0,0 +1,3 @@
+h1 Editing Equipment
+= link_to "Back to Equipment", '/equipment', class: "button"
+=render partial: "form", locals: { equipment: equipment }
```
Need a space here between `=` and `render`

jah2488 commented on app/views/equipment/index.slim:
```diff
@@ -0,0 +1,22 @@
+      td = equipment_item.quantity
+      -if admin_user?
+        td =link_to "Edit", "equipment/#{equipment_item.id}/edit"
+        td =link_to "Remove", "equipment/#{equipment_item.id}", :method => 'delete'
```
I know this is more or less based on existing code, but on closer inspection, its a better practice to use the path helpers so lets change these links to `equipment_path(equipment_item)` and `edit_equipment_path(equipment_item)`

Also lets use the ruby 1.9 style of hashes everywhere so `, method: :delete` would go here.

jah2488 commented on app/views/home/index.html.slim:
```diff
@@ -4,5 +4,6 @@
   -if current_user.admin?
     a.button href='/admin/users' View Users
     a.button href='/horses' Horses
+    a.button href='/equipment' Equipment
```
Lets change these into `link_to` tags using path helpers

`= link_to "Equipment", equipment_path, class: 'button'` etc etc

jah2488 commented on db/seeds.rb:
```diff
@@ -2,11 +2,14 @@
+#Equipment
+Equipment.create!(name: "Saddle", description: "A leather mount for your horse.", quantity: 5)
+Equipment.create!(name: "Carrot and Stick", description: "Boosts speed of horse.", quantity: 3)
+Equipment.create!(name: "Jousting Lance", description: "Make sure guests sign the waver.", quantity: 5)
```
Nice

jah2488 commented:
@VanAnderson I like the idea of moving that into the application controller and all we really have to pass in is the redirection path, so that should work well.

I'm okay with making that abstraction in this PR and once you've pushed the code for that and let me take a look, I can merge this in.

VanAnderson commented:
@jah2488 Good advice! I made sure everything in the views is using link helpers and path helpers, fixed the space, pulled the restrict_non_admin to the application controller, and added some more tests to make sure non-admins are turned away from admin only actions.

As a side bar: I just realized that Pundit is listed as a project gem for authorization. Once we get the barebones framework running, it might be a good idea to have a trello card that switches over all of the authorization over to using Pundit instead of our own hand rolled stuff.

jah2488 commented:
👍🏻
